### PR TITLE
Better handling of gateways in `cloud preapre`

### LINKED ISSUE #1501: subctl cloud prepare must also open the ESP/AH protocols for all nodes (gateways and workers)
* Submariner, when using IPSEC, and Private to Private IP connectivity with no-nat is detedted
will attempt to use ESP/AH protocols for communication (more performant).
If those protocols aren't open for gateways, connecitivty will not work.

* When not using dedicated gateways (load-balancer mode), all the worker nodes need to have NATT/NAT-discovery/ESP/AH open for the same reasons.

### PR DESCRIPTION
**Pull request by mkolesnik**
As reported on #1501, for LB gateways we want to open the ipsec & nat discovery ports. For dedicated gateways, we also want ESP&AH protocols accepted on the gateway SG to support private-to-private tunnelling.

Depends on #1500 
Resolves: #1501

### CODE REVIEW

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 1]
```diff
@@ -65,16 +56,28 @@ func prepareAws(cmd *cobra.Command, args []string) {
 		},
 	}
 
+	// For load-banalced gateways we want these ports open internally to facilitate private-ip to pivate-ip gateways communications.
+	if gateways == 0 {
+		input.InternalPorts = append(input.InternalPorts,
+			api.PortSpec{Port: nattPort, Protocol: "udp"},
+			api.PortSpec{Port: natDiscoveryPort, Protocol: "udp"},
```
Comment (mangelajo) [inline pkg/subctl/cmd/cloud/prepare/aws.go]:
```suggestion
			api.PortSpec{Port: natDiscoveryPort, Protocol: "udp"},
			// ESP & AH protocols are used for private-ip to private-ip gateway communications
			{Port: 0, Protocol: "50"},
			{Port: 0, Protocol: "51"},
```

Comment (mangelajo) [reply pkg/subctl/cmd/cloud/prepare/aws.go]: Here it's also necessary, if private-ip to private-ip connectivity is possible, and no nat is detected, the gateway will attempt to use ESP/AH

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 2]
```diff
@@ -65,16 +56,28 @@ func prepareAws(cmd *cobra.Command, args []string) {
 		},
 	}
 
+	// For load-banalced gateways we want these ports open internally to facilitate private-ip to pivate-ip gateways communications.
```
Comment (mangelajo) [inline pkg/subctl/cmd/cloud/prepare/aws.go]:
```suggestion
	// For load-balanced gateways we want these ports open internally to facilitate private-ip to pivate-ip gateways communications.
```

**Review by mangelajo — CHANGES_REQUESTED** [review]
small comments, linters seem to be upset 

Thanks for handling this.

File: pkg/subctl/cmd/cloud/prepare/aws.go (hunk 1 above)
Comment (mkolesnik) [reply pkg/subctl/cmd/cloud/prepare/aws.go]: these are internal ports between the submariner gateways inside the same cluster, is ESP/AH expected in this scenario? i thought only for gateway-to-gateway external communications

Comment (mkolesnik) [reply pkg/subctl/cmd/cloud/prepare/aws.go]: ah nvm i see your point

**Review by mangelajo — APPROVED** [review]

**Comment by mangelajo** [comment]
:+1:

**Review by tpantelis — APPROVED** [review]

**Pull request merged**